### ISSUE TITLE
Bulk renaming files by 'rename' action moved files in a wrong destination

### ISSUE THREAD

**Issue by nickjj** [lambdalisue/fern.vim#234]
I'm not 100% sure about this but with Vim 8.1 and the latest master version of Fern the following just happened:

- I had 8 jpg files sitting in a directory
- I marked them with Fern
- I issued an action to rename them in a Vim buffer
- I moved them into a different directory that needed root access

The result was no error, but now the files are gone from the source location and they are not in the destination. **In the end this resulted in data loss**.

Edits:

Turns out the files were moved to a directory that was neither the source or destination. I'm not really sure how they ended up there, but they were 1 directory back of the source. The destination in this case wasn't inside of the source.

I've also had related issues 3 times now over the last month where I've had files moved (with the regular rename command) 1 directory back from where it was supposed to be when I was moving a file from `a/foo/test.md` to `b/foo/test.md`. In this case all file permissions were owned by the same user, not root.

**Comment by lambdalisue** [comment]
Fern use `mv` in Linux or `move` (cmd.exe) in Windows and it never remove file/directory by itself (it may overwrite but then the file won't be overwritten).

https://github.com/lambdalisue/fern.vim/blob/master/autoload/fern/scheme/file/shutil.vim#L60-L82
https://github.com/lambdalisue/fern.vim/blob/master/autoload/vital/_fern/Async/File.vim#L112-L145

I'm not sure but I think `mv` in Linux is an atomic operation at least thus I've no idea why you've faced the data loss.

Which OS do you use?

**Comment by nickjj** [comment]
Did you check the edit? It ended up being moved 1 directory back of the source.

I am using Ubuntu 20.04.

**Comment by lambdalisue** [comment]
> Did you check the edit? It ended up being moved 1 directory back of the source.

I don't really get the meaning of the "back of the source" Did you mean that the files were moved to the parent directory of the source? So no data loss?

**Comment by nickjj** [comment]
Yes the parent directory of the source, but that wasn't the destination I supplied.

**Comment by lambdalisue** [comment]
And you've said 

> "Bulk renaming files with the Vim buffer without proper permission loses the files"

but you've said

>  In this case all file permissions were owned by the same user, not root.

as well so I'm confusing. So the file permissions are related to the issue?

It would be nice if you can prepare a step by step procedure to reproduce the issue in my environment while it sounds really complicated and I'm not sure I get the issue or the situation correctly.

**Comment by nickjj** [comment]
Yeah it's happened under 2 different scenarios. Once with different permissions during a bulk rename and a few times without the same permissions while renaming a single file.

I'll try to reproduce it the next time I'm renaming things.

**Comment by nickjj** [comment]
Just a heads up this happened twice over the last 15 days but for the few times I move files with fern, I still haven't been able to reproduce it. In the times I've paid extra close attention to any move commands, it didn't happen.

**Comment by lambdalisue** [comment]
> In the times I've paid extra close attention to any move commands, it didn't happen.

It's sad... Thanks for reporting your current situation anyway 👍

**Comment by nickjj** [comment]
Maybe we should close this and if it happens again, I'll re-comment?

Because I still haven't had it happen again.

**Comment by nickjj** [comment]
I was able to pin point the problem. It was a user error.

Turns out I have `_drafts/podcast/` and `_drafts/interviews/` directories and I often move files from either one of these to `_posts/podcast/` or `/_posts/interviews/`.

The issue was sometimes I thought I created a file in `_drafts/podcast/foo.md` and moved it to `_posts/podcast/foo.md` but really what happened was I created it in `_drafts/foo.md` and moved it to `_posts/foo.md`. So then I came to an incorrect conclusion that somehow the `podcast/` directory got removed during the file move operation when really I created the file in the wrong location initially.

I only did this a few times which is why I couldn't reproduce it in an isolated way.
